MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs

```diff
@@ -0,0 +1,217 @@
+        var role = string.Empty;
+        int? sortOrder = null;
+        string? imageUrl = null;
+
+        using var subtree = reader.ReadSubtree();
+        subtree.MoveToContent();
+        subtree.Read();
+
+        while (subtree is { EOF: false, ReadState: ReadState.Interactive })
+        {
+            if (subtree.NodeType != XmlNodeType.Element)
+            {
+                subtree.Read();
+                continue;
+            }
+
+            switch (subtree.Name)
+            {
+                case "name":
+                case "Name":
+                    name = subtree.ReadElementContentAsString();
+                    break;
+                case "role":
+                case "Role":
+                    var roleValue = subtree.ReadElementContentAsString();
+                    if (!string.IsNullOrWhiteSpace(roleValue))
+                    {
+                        role = roleValue;
+                    }
+
+                    break;
+                case "type":
+                case "Type":
+                    Enum.TryParse(subtree.ReadElementContentAsString(), true, out type);
+                    break;
+                case "order":
+                case "sortorder":
+                case "SortOrder":
+                    if (int.TryParse(
+                        subtree.ReadElementContentAsString(),
+                        NumberStyles.Integer,
+                        CultureInfo.InvariantCulture,
+                        out var intVal))
+                    {
+                        sortOrder = intVal;
+                    }
+
+                    break;
+                case "thumb":
+                    var thumb = subtree.ReadElementContentAsString();
+                    if (!string.IsNullOrWhiteSpace(thumb))
+                    {
+                        imageUrl = thumb;
+                    }
+
+                    break;
+                default:
+                    subtree.Skip();
+                    break;
+            }
+        }
+
+        if (string.IsNullOrWhiteSpace(name))
+        {
+            return null;
+        }
+
+        return new PersonInfo
+        {
+            Name = name.Trim(),
+            Role = role,
+            Type = type,
+            SortOrder = sortOrder,
+            ImageUrl = imageUrl
+        };
+    }
+
+    /// <summary>
+    /// Used to split names of comma or pipe delimited genres and people.
+    /// </summary>
+    /// <param name="reader">The <see cref="XmlReader"/>.</param>
+    /// <returns>IEnumerable{System.String}.</returns>
+    public static IEnumerable<string> GetStringArray(this XmlReader reader)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+        var value = reader.ReadElementContentAsString();
+
+        // Only split by comma if there is no pipe in the string
+        // We have to be careful to not split names like Matthew, Jr.
+        var separator = !value.Contains('|', StringComparison.Ordinal)
```

Comment:
The rest of this method seems a bit inefficient

Reply:
Yeah, this is an ideal case for spans, but we don't have a `Split` method for multiple needles.
One solution is to do a pure fallback of `|` -> `;` -> `,` for separators, which is technically a breaking change, but I doubt there's a ton of files out there using both.

Alternatively, we would have to make another extension method, but imo that's a big enough change to warrant a separate PR.